### PR TITLE
docs: add csrf blog

### PR DESCRIPTION
**Pull request by yzeng25**
Changes:

Add the blog for CSRF plugin.

### CODE REVIEW

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+```shell
+curl -i http://127.0.0.1:9080/hello -X POST
+```
+
+Apache APISIX 会拦截该请求并返回 `401` 错误。在返回的头部中会发现设置了一个 Cookie，如果没有配置插件的 `name` 字段的话，Cookie 内部应该为默认值 `apisix-csrf-token=....` 。这就是 CSRF 插件生成的 CSRF token。在请求中，需要确保请求携带该 Cookie 并且在请求头写入该 token。
+
+```shell
+HTTP/1.1 401 Unauthorized
+Set-Cookie: apisix-csrf-token=eyJyYW5kb20iOjAuNjg4OTcyMzA4ODM1NDMsImV4cGlyZXMiOjcyMDAsInNpZ24iOiJcL09uZEF4WUZDZGYwSnBiNDlKREtnbzVoYkJjbzhkS0JRZXVDQm44MG9ldz0ifQ==;path=/;Expires=Mon, 13-Dec-21 09:33:55 GMT
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md]: The token in the code is too long, it is suggested that the code formatting, or use other words to indicate what this code means.

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+`csrf` 插件基于 `Double Submit Cookie` 方案实现。根据 [RFC 7231#section-4.2.1](https://datatracker.ietf.org/doc/html/rfc7231.html#section-4.2.1) 的定义，我们把 `GET`、`HEAD` 和 `OPTIONS` 这三种方法称为安全方法。按照这一约定，`csrf` 插件会直接放行这三种方法，但会对其他的方法做检查并拦截其中的不安全请求。
+
+为了抵御 CSRF 攻击，我们需要制造一个无法伪造的令牌或标识符，并且保证这个不会与攻击者的请求一起发送。用户需要将 `csrf` 插件依赖的 token 携带在请求头，token 使用密钥进行签名计算。这样就保证了 token 无法被他人伪造，从而保证了 API 的安全。
+
+![插件工作流程](https://static.apiseven.com/202108/1645605178661-7c0bc3bc-9792-43fd-b3f6-b01c0f6b24db.png)
+
+在路由中开启 `csrf` 插件后，访问该路由的所有请求响应中都会包含携带了 `csrf token` 的 Cookie。
+
+用户需要在对该路由的不安全请求中携带这一 Cookie，并在请求头添加额外的字段携带 Cookie 的内容。字段为插件配置中的 `name` 值，这样的请求才能通过 CSRF 插件的校验。
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md]:
```suggestion
用户需要在对该路由的不安全请求中携带这一 Cookie，并在请求头添加额外的字段用来携带 Cookie 的内容。字段为插件配置中的 `name` 值，这样的请求才能通过 CSRF 插件的校验。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+const instance = axios.create({
+  headers: {'apisix-csrf-token': token}
+});
+```
+
+如果 Cookie 中的 token 和请求头中的 token 不一致，请求会被 `csrf` 插件拦截，示例如下：
+
+```shell
+curl -i http://127.0.0.1:9080/hello -X POST -H 'apisix-csrf-token: differenteyJyYW5kb20iOjAuNjg4OTcyMzA4ODM1NDMsImV4cGlyZXMiOjcyMDAsInNpZ24iOiJcL09uZEF4WUZDZGYwSnBiNDlKREtnbzVoYkJjbzhkS0JRZXVDQm44MG9ldz0ifQ==' -b 'apisix-csrf-token=eyJyYW5kb20iOjAuNjg4OTcyMzA4ODM1NDMsImV4cGlyZXMiOjcyMDAsInNpZ24iOiJcL09uZEF4WUZDZGYwSnBiNDlKREtnbzVoYkJjbzhkS0JRZXVDQm44MG9ldz0ifQ=='
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md]:
```suggestion
curl -i http://127.0.0.1:9080/hello -X POST -H 'apisix-csrf-token: ${apisix-csrf-token}' -b 'apisix-csrf-token= ${apisix-csrf-token}'
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+```
+
+```shell
+HTTP/1.1 401 Unauthorized
+Set-Cookie: apisix-csrf-token=eyJyYW5kb20iOjAuNjg4OTcyMzA4ODM1NDMsImV4cGlyZXMiOjcyMDAsInNpZ24iOiJcL09uZEF4WUZDZGYwSnBiNDlKREtnbzVoYkJjbzhkS0JRZXVDQm44MG9ldz0ifQ==;path=/;Expires=Mon, 13-Dec-21 09:33:55 GMT
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md]:
```suggestion
Set-Cookie: apisix-csrf-token= ${apisix-csrf-token};path=/;Expires=Mon, 13-Dec-21 09:33:55 GMT
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+{"error_msg":"csrf token mismatch"}
+```
+
+最后使用 `curl`验证正常的访问：
+
+```shell
+curl -i http://127.0.0.1:9080/hello -X POST -H 'apisix-csrf-token: eyJyYW5kb20iOjAuNjg4OTcyMzA4ODM1NDMsImV4cGlyZXMiOjcyMDAsInNpZ24iOiJcL09uZEF4WUZDZGYwSnBiNDlKREtnbzVoYkJjbzhkS0JRZXVDQm44MG9ldz0ifQ==' -b 'apisix-csrf-token=eyJyYW5kb20iOjAuNjg4OTcyMzA4ODM1NDMsImV4cGlyZXMiOjcyMDAsInNpZ24iOiJcL09uZEF4WUZDZGYwSnBiNDlKREtnbzVoYkJjbzhkS0JRZXVDQm44MG9ldz0ifQ=='
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/02/23/csrf.md]: ditto

File: website/blog/2022/02/23/csrf.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+```shell
+curl -i http://127.0.0.1:9080/hello -X POST
+```
+
+Apache APISIX will intercept the request and return a `401` error. In the returned header you will find a Cookie set, which should be the default value `apisix-csrf-token=....` inside the Cookie if the name field of the plugin is not configured . This is the CSRF token generated by the CSRF plugin. In the request, you need to make sure that the request carries this Cookie and that the token is written in the request header.
+
+```shell
+HTTP/1.1 401 Unauthorized
+Set-Cookie: apisix-csrf-token=eyJyYW5kb20iOjAuNjg4OTcyMzA4ODM1NDMsImV4cGlyZXMiOjcyMDAsInNpZ24iOiJcL09uZEF4WUZDZGYwSnBiNDlKREtnbzVoYkJjbzhkS0JRZXVDQm44MG9ldz0ifQ==;path=/;Expires=Mon, 13-Dec-21 09:33:55 GMT
```
Comment (guitu168) [inline website/blog/2022/02/23/csrf.md]: Please refer to Chinese for modification.

**Review by guitu168 — CHANGES_REQUESTED** [review]

**Review by bzp2010 — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**